Implement latest_trade for the Kraken exchange

sdeslauriers opened the pull request:
This commit implements the `latest_trade` method for the Kraken exchange. In addition, it improves error handling of the `_public_query` method.

sdeslauriers committed:
Implement latest_trade for the Kraken exchange

This commit implements the latest_trade method for the Kraken
exchange. In addition, it improves error handling of the
_public_query method.

gagdiez commented on exchanges/tests/test_kraken.py:
```diff
@@ -34,3 +34,18 @@ def test_server_time(self):
+        # Get the latest trade. I cannot find a way to validate
+        # the result, so I just assume that if it returns something
+        # everything is ok.
+        trade = kraken.latest_trade('XETHZEUR')
```
you know that the result must be of type Trade

gagdiez commented on exchanges/kraken.py:
```diff
@@ -25,15 +30,20 @@ def order_info(self, order_id):
         # OK.
         url = '{}public/{}'.format(self.url, query_name)
-        response = requests.get(url, timeout=self.timeout)
+        response = requests.post(url, data=data, timeout=self.timeout)
```
Question: Why do we change from get to post? All the public methods are safe, and only used to retrieve data. That is correctly reflected with a "get" request. "Post" should be used only with request that will have some side-effect and need to be secured.

sdeslauriers replied:
The documentation says we should be able to query public methods with GET or POST, but I get an `Invalid parameters` error with GET. 

> "Post" should be used only ...

I disagree, there is no downside to using POST.

gagdiez replied:
I agree, there's no downside, but the "get" method expresses better what we want to do: to get public data. And I'm not talking only about the action of "getting something", I'm talking about the action of simply asking for something public, with no server side-effect, and which doesn't need any security measure.

Respect to the Invalid parameters error.. requests.get(url, params=data) should work, I can try if you want.

sdeslauriers replied:
Your are right, works with the keyword `params`. Strange that get accepts `data`, but does not seem to use it.

gagdiez reviewed (commented):
- Not sure about using the post method in public_queries. - test_latest_ticker could be slightly improved.

sdeslauriers committed:
Verify that latest_trade returns a Trade

gagdiez reviewed (commented):
Still not sure about post.

sdeslauriers committed:
Change POST to GET request in _public_query

While both GET and POST can be used for public queries, GET
is more coherent with the objective of _public_query. Changed
POST to GET.
